volume-group-operator/controllers/volumegroup_controller.go

```diff
@@ -70,3 +248,69 @@ func (r *VolumeGroupReconciler) SetupWithManager(mgr ctrl.Manager, cfg *config.D
 		For(&volumegroupv1.VolumeGroup{}).
 		WithEventFilter(pred).Complete(r)
 }
+
+func (r *VolumeGroupReconciler) updateVolumeGroupStatusError(
+	instance *volumegroupv1.VolumeGroup,
+	logger logr.Logger,
+	message string) error {
+	instance.Status.Error = &volumegroupv1.VolumeGroupError{Message: &message}
+	err := r.updateVolumeGroupStatus(instance, logger)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *VolumeGroupReconciler) updateVolumeGroupStatus(instance *volumegroupv1.VolumeGroup, logger logr.Logger) error {
+	if err := r.Client.Status().Update(context.TODO(), instance); err != nil {
+		logger.Error(err, "failed to update status")
+
+		return err
+	}
+	return nil
+}
+
+func (r *VolumeGroupReconciler) deleteVolumeGroup(logger logr.Logger, volumeGroupId string, secrets map[string]string) error {
+	c := volumegroup.CommonRequestParameters{
+		VolumeGroupID: volumeGroupId,
+		Secrets:       secrets,
+		VolumeGroup:   r.VolumeGroupClient,
+	}
+
+	volumeGroup := volumegroup.VolumeGroup{
+		Params: c,
+	}
+
+	resp := volumeGroup.Delete()
+
+	if resp.Error != nil {
+		logger.Error(resp.Error, "failed to delete volume group")
+		return resp.Error
+	}
+
+	return nil
+}
+
+func (r *VolumeGroupReconciler) createVolumeGroup(volumeGroupName string, parameters, secrets map[string]string) *volumegroup.Response {
```

Comment:
```suggestion
func (r *VolumeGroupReconciler) createVolumeGroupResponse(volumeGroupName string, parameters, secrets map[string]string) *volumegroup.Response {
```

Reply:
the function creates the vg on storage, not creating the response.

Reply:
oh sorry right, my bad